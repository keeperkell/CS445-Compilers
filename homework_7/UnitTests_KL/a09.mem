Func: input returns type int [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 1  Func: output returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type int [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 2  Func: inputb returns type bool [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 3  Func: outputb returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type bool [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 4  Func: inputc returns type char [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 5  Func: outputc returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type char [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 6  Func: outnl returns type void [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 7  Var: x of type char [mem: Global loc: 0 size: 1] [line: 2]
Sibling: 8  Func: main returns type void [mem: Global loc: 0 size: -2] [line: 4]
.   Child: 1  Compound [mem: None loc: 0 size: -3] [line: 5]
.   .   Child: 0  Var: y of type char [mem: Local loc: -2 size: 1] [line: 6]
.   .   Child: 1  Assign: <- of type char [line: 8]
.   .   .   Child: 0  Id: x of type char [mem: Global loc: 0 size: 1] [line: 8]
.   .   .   Child: 1  Const 'X' of type char [line: 8]
.   .   Sibling: 1  Assign: <- of type char [line: 9]
.   .   .   Child: 0  Id: y of type char [mem: Local loc: -2 size: 1] [line: 9]
.   .   .   Child: 1  Const 'Y' of type char [line: 9]
.   .   Sibling: 2  Call: outputc of type void [line: 10]
.   .   .   Child: 0  Id: x of type char [mem: Global loc: 0 size: 1] [line: 10]
.   .   Sibling: 3  Call: outnl of type void [line: 11]
.   .   Sibling: 4  Call: outputc of type void [line: 12]
.   .   .   Child: 0  Id: y of type char [mem: Local loc: -2 size: 1] [line: 12]
.   .   Sibling: 5  Call: outnl of type void [line: 13]
Offset for end of global space: -1
Number of warnings: 0
Number of errors: 0
