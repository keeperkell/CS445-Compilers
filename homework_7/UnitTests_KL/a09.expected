Number of warnings: 0
Number of errors: 0
------------------------------------
Loading file: UnitTests/a09.tm
X
Y
Bye.
